modules/login/php/passwordexpiry.class.inc

```diff
@@ -98,15 +99,32 @@ class PasswordExpiry extends \NDB_Form
         }
 
         if (!$this->_user->isPasswordDifferent($plaintext)) {
-            $this->tpl_data['error_message'] = 'You cannot keep the same password';
-            return array('password' => 'You cannot keep the same password');
+            $error_msg = "You cannot keep the same password.";
+            $this->tpl_data['error_message'] = $error_msg;
+            return array(
+                'password' => $error_msg
+            );
+        }
+
+        // Ensure that the password is not the user's email
+        // Otherwise an exception is thrown in User.class::updatePassword
+        // TODO This validation should be done on the front-end too.
+        if ($plaintext == $data['Email']) {
+            $error_msg = "You cannot use your email as your password.";
+            $this->tpl_data['error_message'] = $error_msg;
+            return array(
```

Comment:
There's no reason to split this across multiple lines anymore, now that it doesn't exceed the line length (similarly below.)